recordset.go

```diff
@@ -102,6 +103,24 @@ func (rcs *Recordset) IsActive() bool {
 	return rcs.active.Get()
 }
 
+var ErrRecordsetClosed = errors.New("aerospike: Recordset closed")
+
+// Read reads the next record from the Recordset. If the Recordset has been
+// closed, it returns ErrRecordsetClosed.
+func (rcs *Recordset) Read() (record *Record, err error) {
+	var ok bool
+	select {
+	case <-rcs.cancelled:
+	case record, ok = <-rcs.Records:
+	case err, ok = <-rcs.Errors:
```

Comment:
It's fine if there are records left in either chan. After Close is called, we can assume the recordset will be discarded, the channels GC'd, and with it the records (or errors) in them. So, the channels will be closed, the recordset GC'd, and the world will move on just fine.

Checking for cancelled being closed is the easiest way to determine if the recordset is closed since it's a direct side-effect of Close being called, before any of the Records or Errors channels are closed. As such, it's also the easiest way to immediately bail out of reading off the records/errors channels.

The resources are automatically deallocated by calling Close and allowing the garbage collector to do its job. If the Aerospike client is retaining the same resources elsewhere beyond the recordset's lifetime, I would say that needs to be fixed: once the recordset is closed, that operation is done, the channels should be discarded, and the recordset is no longer valid for use.